_rules/inline-link-in-paragraph-has-distinguishable-style-on-hover-and-focus-548868.md

```diff
@@ -0,0 +1,207 @@
+---
+id: 548868
+name: Inline link in paragraph has distinguishable style on hover and focus
+rule_type: atomic
+
+description: |
+  This rule checks that links that are embedded in a paragraph, list or cell have a style on hover and focus that distinguishes them from the surrounding text
+
+accessibility_requirements: # Remove whatever is not applicable
+  wcag20:1.4.1: # Use of Color (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+
+acknowledgements:
+  authors:
+    - Brian Bors
+    - Carlos Duarte
+  previous authors:
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to any HTML element which:
+
+- has the [semantic role][] of ['link'][link] or a [semantic role][] that inherits from the ['link'][link] role; and
+- is a [descendant][] in the [flat tree][] of a `p`, `li` or `td` HTML elements or elements with a [semantic role][] of ['listitem'][listitem] or ['cell'][cell] (with the closest [ancestor][] in the [flat tree][] being henceforth referred to as the _ancestor_ element); and
+- has [descendant][] [text nodes][text node] in the [flat tree][] that are [visible][]; and
+- the _ancestor_ element has other [descendant][] [visible][] [text nodes][text node] in the [flat tree][].
+
+## Expectation
+
+Each target element has one of the following:
+
+- a [foreground color][] different from the [foreground color][] of the other descendant visible text nodes of the same _ancestor_ element (if all have the same [foreground color][]), that has at least a 3:1 [contrast ratio](https://www.w3.org/TR/WCAG21/#dfn-contrast-ratio) and a [distinguishing style][] both when the target element [gains focus][focused] and the target element is [hovered][]; or
```

Comment:
What is "other descendant visible text nodes"? What exactly does this mean? Remember this isn't "color" but "colors". Text nodes have multiple foreground colors, including all anti-aliased pixels. Do you want all colors from the test target to have a 3:1 contrast with that of all other text nodes? What if there are other links?

Reply:
Good arguments. But the revision made it harder to read. Will appreciate further comments after completing the other changes.